src/lib/ReactCore_Internal.re

```diff
@@ -164,97 +164,139 @@ module Make = (Implementation: HostImplementation) => {
       willReceiveProps: ({state}) => state,
       didMount: defaultDidMount,
       didUpdate: defaultDidUpdate,
-      willUnmount: (_self) => (),
+      willUnmount: _self => (),
       willUpdate: defaultWillUpdate,
       shouldUpdate: defaultShouldUpdate,
-      render: (_self) => assert false,
+      render: _self => assert false,
       initialState: () => (),
       reducer: (_action, _state) => NoUpdate,
-      printState: (_state) => "",
+      printState: _state => "",
       handedOffInstance: ref(None),
       key
-    }
+    };
   };
 
   /*** Tail-recursive functions on lists */
   module ListTR = {
-    let useTailRecursion = (l) =>
+    let useTailRecursion = l =>
       switch l {
       | [_, _, _, _, _, _, _, _, _, _, ..._] => true
       | _ => false
       };
-    let concat = (list) => {
+    let concat = list => {
       let rec aux = (acc, l) =>
         switch l {
         | [] => List.rev(acc)
         | [x, ...rest] => aux(List.rev_append(x, acc), rest)
         };
-      useTailRecursion(list) ? aux([], list) : List.concat(list)
+      useTailRecursion(list) ? aux([], list) : List.concat(list);
     };
     let map = (f, list) =>
       useTailRecursion(list) ?
         List.rev_map(f, List.rev(list)) : List.map(f, list);
     let rev_map3 = (f, list1, list2, list3) => {
-      let rec aux = (acc) =>
+      let rec aux = acc =>
         fun
         | ([], [], []) => acc
         | ([x1, ...nextList1], [x2, ...nextList2], [x3, ...nextList3]) =>
           aux([f((x1, x2, x3)), ...acc], (nextList1, nextList2, nextList3))
         | _ => assert false;
-      aux([], (list1, list2, list3))
+      aux([], (list1, list2, list3));
     };
     let map3 = (f, list1, list2, list3) =>
       rev_map3(f, List.rev(list1), List.rev(list2), List.rev(list3));
   };
 
   /*** Log of operations performed to update an instance tree. */
   module UpdateLog = {
+    type subtreeChange =
+      | Nested
+      | NoChange
+      | PrependElement(renderedElement)
+      | ReplaceElements(renderedElement, renderedElement);
+    type instanceUpdate =
+      | InstanceUpdate(
+                        instance('state, 'action, 'elementType),
+                        instance('state, 'action, 'elementType)
+                      ): instanceUpdate;
     type update = {
```

Comment:
I think this should be `instanceUpdate`. My line of thinking is that an entry is sort of a command (verb) `Update the Instance` and it contains a payload (a noun) `instanceUpdate`.